src/scalar.rs

```diff
@@ -38,7 +38,7 @@
 //! res = a.square(); // Performs a^2 (mod l).
 //! res = -&a; // Performs Negation over the modulo l.
 //! 
-//! // Dividing by two even Scalars is recommended through the `Half`
+//! // Dividing even Scalars by two is recommended through the `Half`
 //! // trait implmementation since it's much faster.
```

Comment:
I think there is a use of dividing here, we should expand on why its faster or give examples.

Reply:
I don't think it's needed to explain why this will be faster since relies on complex codes at some points. But we can discuss it.